test/unit/controllers/Irc_EntryController.test.js

```diff
@@ -17,15 +17,31 @@ describe('INTEGRATION Irc_EntryController', () => {
       });
 
       it('should be able to process update the centre with heartbeat information', () =>
+        expect(controller.process_heartbeat(fake_json)
+          .then(() => Centres.findOne({name: fake_json.centre})))
+          .to.eventually.contain({
+            female_in_use: fake_json.female_occupied,
+            female_out_of_commission: fake_json.female_outofcommission,
+            male_in_use: fake_json.male_occupied,
+            male_out_of_commission: fake_json.male_outofcommission
+          })
+      );
+      describe('with heartbeat timestamp', () => {
+        let clock;
+        before(() => {
+          clock = sinon.useFakeTimers();
+        });
+        after(() => {
+          clock.restore();
+        });
+        it('should update the centre heartbeat timestamp on processing an update', () => {
           expect(controller.process_heartbeat(fake_json)
-	    .then(() => Centres.findOne({name: fake_json.centre})))
+            .then(() => Centres.findOne({name: fake_json.centre})))
             .to.eventually.contain({
-              female_in_use: fake_json.female_occupied,
-              female_out_of_commission: fake_json.female_outofcommission,
-              male_in_use: fake_json.male_occupied,
-              male_out_of_commission: fake_json.male_outofcommission
+              heartbeat_recieved: new Date
```

Comment:
@chrisns I'm not sure what you mean. Should I stub out `Date` to return a value other than a timestamp? Surely that is a contrived test, because I'm only testing that the model contains a `heartbeat_recieved` attribute with *a* value.